module/VuFind/src/VuFind/Form/Form.php

```diff
@@ -1135,4 +1157,29 @@ public function getFormId(): string
     {
         return $this->formConfig['id'] ?? '';
     }
+
+    /**
+     * Validates prefill data and returns only the prefill values for enabled fields
+     *
+     * @param array $prefill Prefill data
+     *
+     * @return array
+     */
+    protected function sanitizePrefill(array $prefill): array
+    {
+        foreach ($this->getProtectedFieldNames() as $fieldName) {
```

Comment:
You could probably eliminate this extra foreach loop by incorporating protected fields into the array_filter callback below -- e.g. change that to `return in_array($key, $prefillFields) && !in_array($key, $this->getProtectedFieldNames())`